pallets/gear/src/manager/journal.rs

```diff
@@ -207,9 +208,22 @@ where
 
         if self.check_program_id(&dispatch.destination()) {
             if let Some(gas_limit) = gas_limit {
-                let _ = GasHandlerOf::<T>::split_with_value(message_id, dispatch.id(), gas_limit);
+                GasHandlerOf::<T>::split_with_value(message_id, dispatch.id(), gas_limit)
+                    .unwrap_or_else(|e| {
+                        // # Safty
+                        //
+                        // 1. there is no logic spliting value from the reserved nodes.
+                        // 2. the `gas_limit` has been checked before.
+                        // 3. the `value` of the value node has been checked before.
+                        result::gas_tree(e)
+                    });
             } else {
-                let _ = GasHandlerOf::<T>::split(message_id, dispatch.id());
+                GasHandlerOf::<T>::split(message_id, dispatch.id()).unwrap_or_else(|e| {
+                    // # Safty
```

Comment:
And it's prettier to paste safety comment above gas handler call to allow unwrap_or_else be single-line call